
The field that stores mount options in 'struct mnttab' is mnt_mntopts on
OmniOS and mnt_opts on Linux; adjust.

diff -wpruN '--exclude=*.orig' a~/gio/gunixmounts.c a/gio/gunixmounts.c
--- a~/gio/gunixmounts.c	1970-01-01 00:00:00
+++ a/gio/gunixmounts.c	1970-01-01 00:00:00
@@ -628,7 +628,7 @@ _g_get_unix_mounts (void)
                                              mntent->mnt_dir,
                                              NULL,
                                              mntent->mnt_type,
-                                             mntent->mnt_opts,
+                                             mntent->mnt_mntopts,
                                              is_read_only);
 
       g_hash_table_insert (mounts_hash,
@@ -743,7 +743,7 @@ _g_get_unix_mounts (void)
                                              mntent.mnt_mountp,
                                              NULL,
                                              mntent.mnt_fstype,
-                                             mntent.mnt_opts,
+                                             mntent.mnt_mntopts,
                                              is_read_only);
 
       return_list = g_list_prepend (return_list, mount_entry);
@@ -1168,7 +1168,7 @@ _g_get_unix_mount_points (void)
       mount_point = create_unix_mount_point (device_path,
                                              mntent->mnt_dir,
                                              mntent->mnt_type,
-                                             mntent->mnt_opts,
+                                             mntent->mnt_mntopts,
                                              is_read_only,
                                              is_user_mountable,
                                              is_loopback);
